data/test/scenarios/wml_tests/ScenarioWML/EventWML/events-test_filter_ability.cfg

```diff
@@ -379,6 +379,137 @@
     [/event]
 )}
 
+#####
+# API(s) being tested: [event][filter_attack][filter_special]active=yes
+##
+# Actions:
+# Use the common setup from FILTER_ABILITY_TEST.
+# Add an event with a filter matching Alice's drains ability, but only when it's active.
+# Alice attacks Bob and then Bob attacks Alice, as defined in FILTER_ABILITY_TEST.
+##
+# Expected end state:
+# The filtered event is triggered exactly once.
+#####
+{GENERIC_UNIT_TEST event_test_filter_special_active (
+    {FILTER_ABILITY_TEST}
+    [event]
+        name=attack
+        first_time_only=no
+        [filter_attack]
+            [filter_special]
+                active=yes
+                tag_name=drains
+                value=25
+            [/filter_special]
+        [/filter_attack]
+        {ASSERT ({VARIABLE_CONDITIONAL side_number equals 1})}
+        {ASSERT ({VARIABLE_CONDITIONAL triggers equals 0})}
+        {VARIABLE_OP triggers add 1}
+    [/event]
+    [event]
+        name=turn 2
+        {RETURN ({VARIABLE_CONDITIONAL triggers equals 1})}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [event][filter_attack][filter_special]active=yes
+##
+# Actions:
+# Use the common setup from FILTER_ABILITY_TEST.
+# Add an event with a filter matching Alice's drains ability, but only when it's active.
+# Give Alice the Illuminates ability, which makes it the wrong time of day for her drains ability.
+# The events in FILTER_ABILITY_TEST make Alice attacks Bob and then Bob attack Alice.
+##
+# Expected end state:
+# The filtered event is never triggered.
+#####
+{GENERIC_UNIT_TEST event_test_filter_special_active_inactive (
+    {FILTER_ABILITY_TEST}
+    [event]
+        name=start
+        [object]
+            silent=yes
+            [effect]
+                apply_to=new_ability
+                [abilities]
+                    {ABILITY_ILLUMINATES}
+                [/abilities]
+            [/effect]
```

Comment:
I think it makes more sense to just change to ToD in the test rather than add an illuminates ability.

Reply:
fixed